plugin/cmd/kube-scheduler/app/configurator.go

```diff
@@ -186,3 +191,113 @@ func (sc schedulerConfigurator) Create() (*scheduler.Config, error) {
 
 	return sc.CreateFromConfig(*policy)
 }
+
+func (sc *schedulerConfigurator) SetupPolicyConfigMapEventHandlers(client clientset.Interface, informerFactory informers.SharedInformerFactory) {
+	// selector targets only the scheduler's policy ConfigMap.
+	selector := cache.NewListWatchFromClient(client.CoreV1().RESTClient(), "configmaps", sc.policyConfigMapNamespace, fields.OneTermEqualSelector(api.ObjectNameField, string(sc.policyConfigMap)))
+
+	sharedIndexInformer := informerFactory.InformerFor(&v1.ConfigMap{}, func(client clientset.Interface, resyncPeriod time.Duration) cache.SharedIndexInformer {
+		sharedIndexInformer := cache.NewSharedIndexInformer(
+			selector,
+			&v1.ConfigMap{},
+			resyncPeriod,
+			cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		)
+		return sharedIndexInformer
+	})
+
+	sharedIndexInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs {
+			AddFunc:    nil, // scheduler aborts if its policy ConfigMap does not exist. So, we do not need and "add" handler.
+			UpdateFunc: sc.updatePolicyConfigMap,
+			DeleteFunc: sc.deletePolicyConfigMap,
+		},
+		0,
+	)
+}
+
+// verifyNewPolicyConfig verifies that the new object received by the ConfigMap watch is a valid policy config.
```

Comment:
Verifying the ConfigMap is an interesting idea (and clearly took some thought and work) but I actually think you shouldn't  do it. More specifically, I think it will be confusing to the user if the scheduler is using an old ConfigMap that is cached in memory after the ConfigMap in etcd has changed. Someone who walks up to the cluster won't be able to look at the ConfigMap to know what the current scheduler behavior is. A related concern is that I think it's better to give the error about the invalid ConfigMap and die right when the ConfigMap is changed (to something invalid) and there is probably a human involved. If you let them get out of sync when the validation fails, then if the scheduler later crashes, when it restarts it will give an error but there may not be anyone around to deal with it.

Of course, on startup we should verify the ConfigMap and exit if it doesn't validate. (I assume we're doing that already; I didn't check.)